src/swoop/api/routers/processes.py

```diff
@@ -137,43 +139,44 @@ async def execute_process(
     except KeyError:
         raise HTTPException(status_code=404, detail="Process not found")
 
-    # Generate a UUID
-    action_uuid = uuid.uuid4()
-
-    # Write to object storage
-
-    request.app.state.io.put_object(
-        object_name=f"executions/{action_uuid}/input.json",
-        object_content=json.dumps(payload),
-    )
+    hashed_pl = workflow.hash_payload(payload=payload)
 
     async with request.app.state.readpool.acquire() as conn:
         async with conn.transaction():
-            # Insert a dummy row into payload_cache table
-            # To be removed later
-            pl_uuid = "debeb36c-e09e-41c3-bdc5-596287fe724a"
-
             q, p = render(
-                "INSERT INTO swoop.payload_cache (:values__names) VALUES :values",
-                values=Values(
-                    payload_uuid=pl_uuid,
-                    workflow_version=2,
-                    workflow_name="mirror",
-                ),
+                "SELECT swoop.check_cache(:plhash,\
+                :wf_version::smallint,:wf_name, :invalid::TIMESTAMPTZ)",
+                plhash=hashed_pl,
+                wf_version=workflow.version,
+                wf_name=workflow.name,
+                invalid=datetime.fromisoformat("2023-06-05T15:49:03+00:00"),
```

Comment:
Being able to pass in an invalid date is great, as we can then support a TTL on cache entries as a workflow config parameter in the future. But

1. this is not something we have to support today, so if doing so adds too much complexity (especially if we aren't testing it), then it might make sense to not
2. right now the `invalid_after` time should always be `NULL` because we don't have any config for workflow cache TTLs